distributeaid/aggregated-public-information
# feat(data-ingestion): Upload regions and subregions from needs data to the collections in Strapi

## What changed?
Breaking tasks from PR #66 into smaller pieces to populate Strapi collections. This particular pull request **partially** resolves Issue #48, populating the `Geo.Region` and `Geo.Subregion` collections in Strapi.

Still to do:

- [ ] 1. Populate the remaining Strapi collections that are a "relation" field type for the `NeedsAssessment.Need` collection:
             - `NeedsAssessment.Survey`
             - `Product.Item`
- [ ] 2. Populate all fields in the `NeedsAssessment.Need` collection with the required data.

Thought: We may want to create seperate issues for each of the above noted tasks allowing multiple contributors to work on the final objective.

## How can you test this?

Add the attached [needs-data.json](https://github.com/user-attachments/files/16826155/needs-data.json) file to the `src/scripts/import-needs-assessment-data` folder, setup the `src/scripts/.env` file, then run the dev server in one terminal and `yarn script:import-needs-assessment-data` in another. The logs should show you that 5 regions and 9 subregions were either created or already exist.
